src/MessageHandler/HashFileHandler.php

```diff
@@ -2,14 +2,75 @@
 
 namespace App\MessageHandler;
 
-use App\Message\HashFile;
-
+use Doctrine\ORM\EntityManagerInterface;
+use League\Flysystem\FilesystemInterface;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
 
+use App\Message\HashFile;
+use App\Entity\File;
+use App\Repository\FileRepository;
+
+/**
+ * The handler for file hashing messages
+ */
 class HashFileHandler implements MessageHandlerInterface
 {
-    public function __invoke(HashFile $HashFile)
+    /**
+     * The entity handler.
+     *
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * The file Repository.
+     *
+     * @var FileRepository
+     */
+    private $fileRepository;
+
+    /**
+     * The monolog logger.
+     *
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Constructor for this Controller, to set up default services.
+     *
+     * @param EntityHandler   $entityHandler           The entity handler.
+     * @param FileRepository  $fileRepository          The file Repository.
+     * @param LoggerInterface $datasetFileHasherLogger Name hinted dataset_file_hasher logger.
+     */
+    public function __construct(EntityManagerInterface $entityManager, FileRepository $fileRepository, LoggerInterface $datasetFileHasherLogger)
+    {
+        $this->entityManager = $entityManager;
+        $this->fileRepository = $fileRepository;
+        $this->logger = $datasetFileHasherLogger;
+    }
+
+    /**
+     * Invoke function to hash a file.
+     */
+    public function __invoke(HashFile $hashFile)
     {
-        //
+        $fileId = $hashFile->getFileId();
+        $this->logger->info(sprintf('Processing File with ID: %d', $fileId));
+        $file = $this->fileRepository->find($fileId);
+        if ($file instanceof File) {
+            $filePath = $file->getFilePath();
+            try {
+                $fileHash = hash_file('sha256', $filePath);
+            } catch (\Exception $e) {
+                $this->logger->error(sprintf('Unable to hash file. Message: %s', $e->getMessage()));
+                return;
```

Comment:
I decided to do a return on error, because if the file can't be hashed for whatever reason, there is no need to retry? Whadda you guys think?

Reply:
Do we want to create an Exception class, so that based on the type of error we can either put it in the queue or return void?

Reply:
It just returns a general exception, and creating another class won't do much good. Maybe we should see how it goes and then see if these exceptions need to be retried.